### ISSUE TITLE
Implement an API layer to divide CORE from Disassembler

### ISSUE THREAD

**Issue by ITAYC0HEN** [dod-cyber-crime-center/kordesii#4]
Currently, kordesii is built above IDA-Python and dependent on IDA to achieve its stuff. That, when IDA is basically supplying for Kordesii the information to work on, where Kordesii itself performs the logic.

My suggestion is to use a more flexible and modern design for Kordesii in such a way that an API layer is implemented above CORE functionality. The logic will be implemented without being dependent on a specific disassembler to supply the information.

In this way, the community would be able to implement plugins for other disassemblers such as [Cutter](https://github.com/radareorg/cutter), [Radare2](https://github.com/radare/radare2), Binary Ninja, and GHIDRA.

This method will expand the usability of Kordesii and your great implementation for IDA can be a template or a go-to reference for other plugins.

To sum up, the solution I would like to see will be

```
Cutter        IDA      Radare2      Binja      GHIDRA
    |__________|_________|___________|_________|
                         |
                      API Layer
                         |
                    Kordesii Logic
```

**Comment by DC3-DCCI** [comment]
Thank you for the feedback. Adding disassembler agnosticism is something we are currently looking into. We will welcome contributions for adding support for other disassemblers once the framework implementation is in place.

**Comment by DC3-DCCI** [comment]
We have begun working on a new project to allow the use of disassemblers other than IDA.  The alpha version of this is located at https://github.com/Defense-Cyber-Crime-Center/Dragodis.  Kordesii will be updated to depend on it when the project reaches a 1.0 version.  This issue will remain open until these change is made.

**Comment by dc3-tsd** [comment]
Closing this issue as Rugosa can accomplish this using Dragodis and is the successor to Kordesii, which is now only receiving updates for maintence prior to it being formally deprecated (no formal date has been decided for this).
